### ISSUE TITLE
Alex review updated contracts; recommendations

### ISSUE THREAD

**Issue by trentmc** [oceanprotocol/df-py#145]

**Comment by alexcos20** [comment]
Contracts lgtm, but I have some recommendations:
 - when deploying the contracts, use the following steps:
 - - deploy DFRewards
 - - deploy DFStrategyV1  (taking DFRewards address from above)
 - - add strategy to DFRewards
 - - move ownership of DFRewards to OPFV4Wallet multisig (I will give you the list)
 
 - use an array instead of mapping for live_strategies  (It's more expensive, but I do not expect ten of thousands of strategies). It's far more practical for a dApp to store only DFRewards contract address and fetch strategy addresses on the fly
 - each strategy should have a public int or string, in order for the dApps to match it to a known ABI

**Comment by alexcos20** [comment]
mainnet: 0x0d27cd67c4A3fd3Eb9C7C757582f59089F058167
ropsten: 0x58F76AE5BC7Fe80D2fb2781d92189e6eE6Eb8F76
rinkeby:  0x0e901bC5D49636eC75B3B4fB88238698E5322dE6
polygon: 0x6272E00741C16b9A337E29DB672d51Af09eA87dD
moonbase: 0xd8992Ed72C445c35Cb4A2be468568Ed1079357c8
mumbai: 0x06100AB868206861a4D7936166A91668c2Ce1312
bsc: 0x62012804e638A15a5beC5aDE01756A7C8d0E50Cc
energyweb: 0xB98f46485e8b9206158D8127BAF81Dbfd6139Cef
moonriver: 0x06100AB868206861a4D7936166A91668c2Ce1312

**Comment by trentmc** [comment]
The comment above handles 1 to the extent that this issue can do it.

The [PR just added](https://github.com/oceanprotocol/df-py/pull/157) solves both 2 and 3.

**Comment by trentmc** [comment]
Great to get the work so far merged, hooray!

Is it tested enough? I didn't see any changes to testing code. Kinda surprised me, changes like this to the smart contract should have broke some tests.

And, for sure, we should have a test that it queries the contract for the current list of strategies.

**Comment by trentmc** [comment]
> It might be nice to add a test to query the list of strategies. I can create a PR to add it.
Yes, please do that. It's a really useful feature like Alex had outlined.

Once the new PR is merged, we can close this issue.
